stacks_queues/queue.py

```diff
@@ -15,47 +15,69 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        if self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.front == -1:
```

Comment:
Since you never check after dequeueing an element, whether the queue becomes empty and thus you should move the `front` and `rear` pointers back to -1, this conditional may not always work. Try using the `empty` method you implement below instead.
```suggestion
        if self.empty():
```